.github/workflows/Readme.md

```diff
@@ -0,0 +1,297 @@
+## Workflow Overview
+
+The workflow consists of several jobs that execute sequentially or in parallel, depending on their dependencies.
+
+### PREPARE-job
+
+This initial job sets up the environment and determines whether the workflow should proceed based on the trigger logic.
+
+**Key Steps:**
+
+1. **Checkout Repository**: Uses `actions/checkout@v4` to fetch the latest code with a `fetch-depth` of 2 to ensure sufficient commit history for comparison.
+
+2. **Get Changed Files**: 
+   - Uses `git diff` to identify Dockerfiles (`*.dockerfile`) that have changed between the previous commit and the current one.
+   - Stores the list of changed files in the `files` output variable.
+
+3. **Set Proceed Flag**: 
+   - Checks if no Dockerfiles or multiple Dockerfiles have been modified.
+     - If no Dockerfiles are modified, sets `proceed_valid` to `false` and exits.
+     - If multiple Dockerfiles are modified, sets `proceed_valid` to `false` and exits.
+   - If exactly one Dockerfile is modified, proceeds to check for the required label.
+   - Verifies that the Dockerfile contains `LABEL org.opencontainers.image.compilation=auto`.
+     - If present, sets `proceed_valid` to `true`.
+     - If not, sets `proceed_valid` to `false` and exits.
+
+4. **Parse File Path**: Extracts necessary information from the modified Dockerfile:
+   - `directory`: The directory containing the Dockerfile.
+   - `dockerfile_name`: The base name of the Dockerfile without the extension.
+   - `platform` and `platform_tag`: Determined by checking the `org.opencontainers.image.arch` label in the Dockerfile.
+     - Supports `linux/arm64` (tagged as `arm`) and `linux/amd64` (tagged as `x86`).
+
+5. **Set Current Date**: Captures the current date in `MM-DD` format for use in tagging the Docker image.
+
+6. **Debug Variables**: Outputs the collected variables for verification purposes.
+
+### BUILD-AND-PUSH-job
+
+This job builds the Docker image using Docker Buildx and saves it for later use.
+
+**Key Steps:**
+
+1. **Set Up QEMU**: Uses `docker/setup-qemu-action@v3` to enable cross-platform emulation, allowing builds for different architectures.
+
+2. **Set Up Docker Buildx**: Initializes Docker Buildx with the `docker-container` driver.
+
+3. **Enable BuildKit**: Activates Docker BuildKit by setting `DOCKER_BUILDKIT=1` in the environment, enhancing build performance.
+
+4. **Build Docker Image**:
+   - Uses `docker/build-push-action@v6` to build the Docker image.
+   - Context and Dockerfile path are set based on the modified Dockerfile.
+   - Image is tagged using the format: `{dockerfile_name}-{platform_tag}:{date}`.
+   - Specifies the target platform (`linux/arm64` or `linux/amd64`).
+   - Uses caching mechanisms:
+     - `cache-from: type=gha` to pull cache from GitHub Actions cache.
+     - `cache-to: type=gha,mode=max` to save build cache layers.
```

Comment:
I do wonder if we should limit this given some recipes are very heavy builds (both in compute time and also size).

Reply:
Yes. In the script, I have adjusted the step from "GHA" to local storage. I also updated the readme here.